feat: add eval_config support to RL API client

JannikSt opened the pull request:
Adds missing `eval_config` support to the RL API client.

Changes:
- Add `eval_config` field to `RLRun` model
- Add `eval_config` parameter to `create_run()`
- Wire up `payload["eval"]` in request body

Relates to #256

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds evaluation support to RL runs and a new logs experience.
> 
> - Adds `eval_config` to `RLRun`, plumbs `eval_config` through `RLClient.create_run()` as `payload["eval"]`
> - CLI: introduces `[eval]` config (envs, interval, num_examples, rollouts_per_example, base_model) and corresponding `--eval-*` flags; validates env/eval slugs; surfaces eval settings in run creation output
> - New `prime rl logs` command with `--tail` and `--follow`; cleans output by stripping ANSI and collapsing progress bars; handles rate limiting
> - API client: new `get_logs(run_id, tail_lines)` method used by CLI
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit aa1b26d886917ebea238f805710e14d4e2304c47. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

JannikSt committed:
feat: add eval_config support to RL API client

JannikSt committed:
Remove accidentally committed test files

JannikSt commented on packages/prime/src/prime_cli/api/rl.py:
```diff
@@ -131,6 +133,9 @@ def create_run(
                 payload["run_config"] = run_config
 
+            if eval_config:
+                payload["eval"] = eval_config
```
This is correct - matches the backend API schema. The request field is `eval` (see `CreateRunRequest.eval` in rft/schemas.py:170), while the response returns it as `evalConfig` via alias.

JannikSt committed:
feat: add logs command for RL runs

JannikSt committed:
fix: move time import to top, add rl_config example

JannikSt committed:
feat: add --watch flag and improve log streaming

JannikSt committed:
fix: allow built-in envs like reverse-text, update example

JannikSt committed:
feat: add --eval-* options to rl run command

JannikSt committed:
fix: strip ANSI escape codes from logs output

JannikSt committed:
fix: increase poll interval to 5s, add rate limit handling

JannikSt committed:
fix: filter progress bars from logs output, remove redundant --watch flag

JannikSt committed:
fix: keep 100% progress bar completion lines in logs

JannikSt committed:
fix: address review comments - simplify log follow, warn on unused eval options

JannikSt commented on packages/prime/src/prime_cli/commands/rl.py:
```diff
@@ -273,15 +270,48 @@ def delete_run(
+                    for line in lines[seen_lines:]:
+                        console.print(line)
+                    seen_lines = len(lines)
+                time.sleep(2)
```
Fixed in 309c6e0 - simplified log follow to track by line count instead of hash, no more memory leak or duplicate suppression issues.

JannikSt commented on packages/prime/src/prime_cli/commands/rl.py:
```diff
@@ -273,15 +270,58 @@ def delete_run(
+                        line_hash = hash(line)
+                        if line_hash not in printed_lines:
+                            console.print(line)
+                            printed_lines.add(line_hash)
```
Fixed in 309c6e0 - removed hash-based deduplication, now uses simple line count tracking.

JannikSt commented on packages/prime/src/prime_cli/commands/rl.py:
```diff
@@ -424,10 +485,25 @@ def create_run(
+        if eval_rollouts is not None:
+            parsed_eval_config["rollouts_per_example"] = eval_rollouts
+        if eval_base_model is not None:
+            parsed_eval_config["eval_base_model"] = eval_base_model
```
Fixed in 309c6e0 - now shows a warning when eval options are provided without --eval-envs.

JannikSt commented:
@codex review

JannikSt committed:
fix: handle log rotation in follow mode when tail window is full

JannikSt commented on packages/prime/src/prime_cli/commands/rl.py:
```diff
@@ -273,15 +302,76 @@ def delete_run(
+                        # Print new lines (those beyond what we had before)
+                        # This handles the case where logs grow at the end
+                        if len(new_lines) > len(old_lines):
+                            for line in new_lines[len(old_lines) :]:
```
Fixed in 37b5f64 - now properly detects log rotation by finding the overlap between old and new tail windows, and prints only the truly new lines.

JannikSt committed:
fix: always use overlap detection for log follow to handle fast growth with rotation

JannikSt commented on packages/prime/src/prime_cli/commands/rl.py:
```diff
@@ -273,15 +302,87 @@ def delete_run(
+                        elif len(new_lines) > len(old_lines):
+                            # Logs grew, print new lines at the end
+                            for line in new_lines[len(old_lines) :]:
+                                console.print(line)
```
Fixed in 8fa1d3d - now always uses overlap detection regardless of whether logs grew or shrank, handling both growth and rotation in all cases.

JannikSt committed:
feat: add [eval] section support in TOML config files

JannikSt committed:
fix: improve progress bar filtering to remove empty lines

JannikSt committed:
fix: require owner/name format for environments, remove example config

manveerxyz commented on packages/prime/src/prime_cli/commands/rl.py:
```diff
@@ -449,17 +590,66 @@ def create_run(
+        parsed_eval_config = {
+            "environments": [{"id": env} for env in final_eval_envs],
+        }
+        # CLI options take precedence over config file
```
there is a `from_sources` func that was added to the `BaseConfig` that is meant to handle precedence between config and cli options in a more general purpose way -- ideally we'd leverage that as these if statements are a little messy

manveerxyz commented on packages/prime/src/prime_cli/commands/rl.py:
```diff
@@ -383,11 +501,34 @@ def create_run(
+    eval_envs: Optional[List[str]] = typer.Option(
+        None,
+        "--eval-envs",
+        help="Environments to evaluate on (e.g., 'reverse-text' or 'owner/env')",
```
I think we have to require users to specify it as `owner/env`

JannikSt committed:
fix: use from_sources for eval config merging, require owner/name format

- Use BaseConfig.from_sources for eval config precedence instead of manual if-statements
- Require owner/name format for --eval-envs (same as training environments)
- Rename EvalConfig.eval_base_model to base_model for proper underscore mapping

JannikSt commented:
@codex review
